### PR TITLE
Functionality Migration and Bug Fixes

### PR DESCRIPTION
**Pull request by JasonGould81**
# browserAction
The creation and storage of the browserAction menu has been moved to the background script to allow it to be static. The menu is now always created with icon elements which are either shown or hidden depending on the relevant preference. The menu is still rebuilt each time the preview or previewDelay preferences are changed as those values are stored in closures attached to the menu and cannot be updated otherwise.

# Preference Changes are now Observed
Added a listener for changes to preferences in storage.local. Upon intercepting a valid change, the script reacts appropriately for all migrated functionality. At the moment this is mostly related to the browserAction menu and rotate functionality.

# Preferences Synced
When a preferences held in storage.local is changed it now triggers a message to be sent to the bootstrap code to update its equivalent preference with the new value.  

**NOTE:** This is a one way street due to the nature of embedded WebExtensions. Therefore, preferences which are changed by the bootstrap code **DO NOT** automatically update the equivalent preferences in storage.local. As such, special measures have been taken to query these preferences in the bootstrap code whenever they are needed. This mostly concerns the auto and current preferences.

# Rotate Functionality
Rotate functionality migrated to the WebExtension. Because the shortcuts are still handled by the bootstrap code, and the auto rotate functionality can be toggled by a shortcut, the auto rotate periodic alarm is currently set to run continuously. When the alarm triggers, the rotate is only performed if the auto preference is true. Changes to the autoMinutes preference will restart the timer with the new period value. Likewise, themes switched using the browser action will restart the timer. 

**NOTE:** Themes switched using the shortcut or the main menu or tools submenu **DO NOT** restart the timer as this happens purely in the bootstrap code which is unable to initiate communication with the embedded WebExtension.

# Bug Fixes
A serious case of **"How did that ever work?!!"**. Numerous bugs in bootstrap.js fixed, including user preferences not being removed on uninstall, default preferences being loaded (in a seperate context) after the preferences observer was set causing MANY PersonaSwitcher not defined exceptions, and others. Moved some functions from ui.js to PersonaSwitcher.jsm to facilitate bug fixes in bootstrap.js.

### CODE REVIEW

**Comment by trevermock54** [comment]
It all seems good to me. I've tinkered around with it for about an hour so far (mostly preferences/internationalization stuff) and haven't been able to break anything. Code looks good as well from what I can tell!
-Trever

File: bootstrap.js [hunk 1]
```diff
@@ -1,35 +1,38 @@
...
+var styleSheetService = Cc["@mozilla.org/content/style-sheet-service;1"].getService(Ci.nsIStyleSheetService);
+var uri = Services.io.newURI("chrome://personaswitcher/skin/toolbar-button.css", null, null);
 
-function startup(data, reason) {
-  Cu.import('chrome://personaswitcher/content/PersonaSwitcher.jsm');
-	
-	data.webExtension.startup().then(api => {
-    const {browser} = api;
-    browser.runtime.onMessage.addListener(messageHandler);
-  });
```
Comment (steppy08) [inline bootstrap.js]: Why was this removed and what is it doing?

File: chrome.manifest [hunk 2]
```diff
@@ -1,5 +1,5 @@
-content   personaswitcher		    chrome/content/
-locale    personaswitcher	en-US	chrome/locale/en-US/
+content   personaswitcher		    	chrome/content/
+locale    personaswitcher	en-US		chrome/locale/en-US/
```
Comment (steppy08) [inline chrome.manifest:2]: Is this weird spacing stuff due to your editor?

Comment (JasonGould81) [reply chrome.manifest:2]: Yea, sadly it's in a few places. I can push up a commit just for this if we want, or wait until we go through and clean it up to match whatever we decide the conventions will be.
